### PR TITLE
Homework/Arrays

### PR DESCRIPTION
**Pull request by Jacob-Osho**

### CODE REVIEW

File: firstApp/Arrays/Program.cs [hunk 1]
```diff
@@ -4,6 +4,175 @@ namespace Arrays
...
+                    case SortAlgorithmType.Insertion:
+                        SortByInsertion(myArr);
+                        break;
+                    default:
+                        Console.WriteLine("Entered Wrong Information! ");
+                        break;
+                }
+                switch (decision2)
+                {
+                    case OrderBy.Ascending:
```
Comment (sergey-olefir) [inline firstApp/Arrays/Program.cs:139]: I'd add OrderBy as a param to sort method and made it optional

File: firstApp/Arrays/Program.cs [hunk 2]
```diff
@@ -4,6 +4,175 @@ namespace Arrays
...
+
+        static void Main(string[] args)
+        {
+
+
+            int[] myArr = { 17, 2, 3, 22, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, -16, 1, 18, };
+
+
+            int[] myNewArr = Sort.SortArray(myArr, Sort.SortAlgorithmType.Bubble, Sort.OrderBy.Ascending);
+            for (int i = 0; i < myNewArr.Length; i++)
```
Comment (sergey-olefir) [inline firstApp/Arrays/Program.cs:161]: foreach

**Review by sergey-olefir — CHANGES_REQUESTED** [review]

File: firstApp/Arrays/Program.cs [hunk 3]
```diff
@@ -4,6 +4,175 @@ namespace Arrays
...
+            {
+                int temp2;
+                int[] array = array1;
+                for (int b = 0; b < array.Length - 1; b++)
+                {
+
+
+                    for (int i = 0; i < array.Length - 1; i++)
+                    {
+                        if (i == array.Length - 1)
```
[the remark this answers is not in the document]
Comment (Jacob-Osho) [reply firstApp/Arrays/Program.cs:41]: тупанул... вопросов нет

File: firstApp/Arrays/Program.cs [hunk 4]
```diff
@@ -4,6 +4,175 @@ namespace Arrays
...
+        public static class Sort
+        {
+            private static int[] SortBySelection(int[] array)
+            {
+                int[] array2 = array;
+                int minIndex;
+                int temp;
+                for (int i = 0; i < array2.Length; i++)
+                {
+                    minIndex = Array.FindIndex(array, x => x == array2[i..].Min());
```
[the remark this answers is not in the document]
Comment (Jacob-Osho) [reply firstApp/Arrays/Program.cs]: Solved Everything ! please check it !

File: firstApp/Arrays/Program.cs [hunk 5]
```diff
@@ -4,6 +4,170 @@ namespace Arrays
...
+                    int minIndex = i;
+                    for (int j = i; j < array.Length;j++)
+                    {
+                        if (array[j] < minvalue)
+                        {
+                            minvalue=array[j];
+                            minIndex = j;
+                        }  
+                    }
+                        //Array.FindIndex(array, x => x == array[i..].Min
```
Comment (sergey-olefir) [inline firstApp/Arrays/Program.cs:26]: Don't leave comments. It's a bad habit

File: firstApp/Arrays/Program.cs [hunk 6]
```diff
@@ -4,6 +4,170 @@ namespace Arrays
...
+                for (int i = 0; i < array.Length; i++)
+                {
+                    minvalue = array[i];
+                    int minIndex = i;
+                    for (int j = i; j < array.Length;j++)
+                    {
+                        if (array[j] < minvalue)
+                        {
+                            minvalue=array[j];
+                            minIndex = j;
```
Comment (sergey-olefir) [inline firstApp/Arrays/Program.cs:23]: Try to follow a consistent code style. Spaces between equal sign

File: firstApp/Arrays/Program.cs [hunk 7]
```diff
@@ -4,6 +4,170 @@ namespace Arrays
...
+                    array[i] = array[minIndex];
+                    array[minIndex] = temp;
+                    //Console.Write(array[i] + " "); насамом деле это было сделанно что бы удобнее дебажить
+                }
+                return array;
+            }
+            private static int[] SortByBubble(int[] array)
+            {
+                int temp;
+                for (int b = 0; b < array.Length - 1; b++)
```
Comment (sergey-olefir) [inline firstApp/Arrays/Program.cs:37]: Better use j for nested loops. It's a standart

File: firstApp/Arrays/Program.cs [hunk 8]
```diff
@@ -4,6 +4,170 @@ namespace Arrays
...
+                    counter = i;
+                    if (array[i] > array[i + 1])
+                    {
+                        temp3 = array[i];
+                        array[i] = array[i + 1];
+                        array[i + 1] = temp3;
+
+
+
+
```
Comment (sergey-olefir) [inline firstApp/Arrays/Program.cs:70]: Spaces

**Review by sergey-olefir — CHANGES_REQUESTED** [review]

File: firstApp/Arrays/Program.cs [hunk 9]
```diff
@@ -4,6 +4,170 @@ namespace Arrays
...
+                Bubble,
+                Insertion
+
+            }
+            public enum OrderBy
+            {
+                Ascending,
+                Descending,
+            }
+            public static int[] SortArray(int[] myArr, SortAlgorithmType decision1, OrderBy decision2 = OrderBy.Ascending)
```
Comment (sergey-olefir) [inline firstApp/Arrays/Program.cs:109]: each algorithm should accept order direction. Let's move it.

**Review by sergey-olefir — CHANGES_REQUESTED** [review]

**Comment by Jacob-Osho** [comment]
решил все

**Pull request merged**